app/uk/gov/hmrc/agentservicesaccount/Errorhandler.scala

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.agentservicesaccount
+
+import javax.inject.{Inject, Singleton}
+
+import com.google.inject.name.Named
+import play.api.http.HeaderNames.CACHE_CONTROL
+import play.api.http.HttpErrorHandler
+import play.api.i18n.{Messages, MessagesApi}
+import play.api.mvc.Results._
+import play.api.mvc.{Request, RequestHeader, Result}
+import play.api.{Configuration, Environment, Mode}
+import uk.gov.hmrc.agentservicesaccount.views.html.error_template
+import uk.gov.hmrc.auth.core.{InsufficientEnrolments, NoActiveSession}
+import uk.gov.hmrc.http.{JsValidationException, NotFoundException}
+import uk.gov.hmrc.play.HeaderCarrierConverter
+import uk.gov.hmrc.play.audit.http.connector.AuditConnector
+import uk.gov.hmrc.play.bootstrap.http.FrontendErrorHandler
+import uk.gov.hmrc.play.bootstrap.config.{AuthRedirects, HttpAuditEvent}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class ErrorHandler @Inject() (
+                               val env: Environment,
+                               val messagesApi: MessagesApi,
+                               val auditConnector: AuditConnector,
+                               @Named("appName") val appName: String)(implicit val config: Configuration, ec: ExecutionContext)
+  extends FrontendErrorHandler with AuthRedirects with ErrorAuditing {
+
+  override def onClientError(request: RequestHeader, statusCode: Int, message: String): Future[Result] = {
+    auditClientError(request, statusCode, message)
+    super.onClientError(request,statusCode,message)
+  }
+
+  override def resolveError(request: RequestHeader, exception: Throwable) = {
+    auditServerError(request,exception)
+    exception match {
+      case _: NoActiveSession => toGGLogin(if (env.mode.equals(Mode.Dev)) s"http://${request.host}${request.uri}" else s"${request.uri}")
+      case _: InsufficientEnrolments => Forbidden
+      case _ => super.resolveError(request, exception)
+    }
+  }
+
+  override def standardErrorTemplate(pageTitle: String, heading: String, message: String)(implicit request: Request[_]) = {
+    error_template(
+      Messages("global.error.500.title"),
```

Comment:
These messages are missing from the messages file

Reply:
Good catch! will add it now..